roles/awsenv/tasks/network.yml

```diff
@@ -49,15 +50,15 @@
 
   - name: Set value of new gateway id for route table
     set_fact:
-      nat_gw_id: "{{ nat_gw.nat_gateway.nat_gateway_id }}"
+      nat_gw_id: "{{ nat_gw.nat_gateway_id }}"
     register: gw_change
 
-  when: "{{ has_nat_gw.NatGateways | count < 1 }}"
+  when: has_nat_gw.result | count < 1
 
 - name: Set value of gateway id for route table
```

Comment:
It's not an override AFAIK. First is the creation of the nat gatway if it doesn't exist, and the second sets is for both cases.
If you can suggest better naming, I'm happy to use it.